### DISCUSSION TITLE
Elasticsearch niet langer opensource

### DISCUSSION THREAD

**Discussion by hannekevanderhorst** [Klantinteractie-Servicesysteem/KISS-frontend#87] [General]
Elasticsearch is niet mee opensource. 
Toepassen van elastich gaat dan tegen de principes van Common Ground in.
Wat is de impact op KISS en zijn er alternatieven.
https://opensourceconnections.com/blog/2021/01/15/is-elasticsearch-no-longer-open-source-software/ 

En dit citaat helpt wel: "Those considering Elasticsearch for new projects will have to consider how important they regard the freedoms of a true open source license and perhaps examine alternatives such as Apache Solr (which has no single company controlling contributions) and Vespa (which although not as well known, powers Yahoo and has several innovative new features)."
 
Omdat ElasticSearch in het verleden al niet betrouwbaar was met de belofte open source te blijven, zou ik er niet op durven vertrouwen dat er niet nog meer verandert. SOLR ligt erg voor de hand als vervangende tool: bewezen technologie en echt open source.

https://www.reworked.co/knowledge-findability/not-so-open-any-more-elasticsearch-relicensing-and-implications-for-open-source-search/

Bottom line is dat code van ElasticSearch open is, maar dat ze geen open source licentie hebben volgens de definitie. je mag niet alles met de code doen. Citaat: "if you want to provide Elasticsearch on a SaaS basis, you have to release any code that you use to do this". Dat zijn we bij KISS niet van plan. Maar toch introduceer je een risico dat in de toekomst ook andere activiteiten niet zijn toegestaan met de code. Zoals gezegd Elastic is niet een betrouwbare partner gebleken.

**Comment by hannekevanderhorst** [comment]
Impact KISS
In eerste instantie dachten we dat het niet opensource zijn van Elastic geen invloed zou hebben op ons project. Wij gebruiken een klein onderdeel: meta enige, een methode om meerdere indexen in een verzameld zoekresultaat weer te geven. Dit zit in de Platinum licentie van 125 per maand zie https://www.elastic.co/pricing/. We hebben deze 2 x nodig voor de dev omgeving en voor demodam 250 in de maand dus. Nu horen we dat de on-premise variant van dit onderdeel 10K kost (wat nergens op de site te vinden is)

Dimpact heeft het voornemen om een generiek Zoek & Vind component te realiseren voor de vele toepassingen. In veel klantreizen is zoek & vind functionaliteit nodig. Daarin zal zeker de open source discussie meegenomen worden, want het is één van de Common Ground principes en daarmee uitgangspunt in de Dimpact architectuur.

**Comment by rubenvdlinde** [comment]
Ik heb onder -> https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/issues/187 een uitwerking toegevoegd van wat er zou gebeuren als we dat kleine stuk van de elastic search functionaliteit die we daadwerkelijk gebruiken zouden mee nemen in nieuwe functionaliteit voor de gateway
